modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageExpirationWorker.java

```diff
@@ -31,9 +39,19 @@
 
     private static Log log = LogFactory.getLog(MessageExpirationWorker.class);
     private volatile boolean working = false;
+    /**
+     * hold the expired messages detected in the message flusher delivery rule check for batch delete
+     */
+    public static Set<DeliverableAndesMetadata> expiredMessageSet = new HashSet<>();
```

Comment:
Also, I guess we done need to keep 'DeliverableAndesMetadata'. Having the message ids is enough isn't it?